### PR TITLE
[ADD] New structure for animation controller

### PR DESCRIPTION
**Pull request by Erick9Thor**

### CODE REVIEW

**Review by monikamnc — APPROVED** [review]
Tested and everything looks good to me, Thanks!!
(Amazing the ::byte solution for the bug)

File: Source/src/components/ComponentAnimation.cpp [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+
+#include "ComponentAnimation.h"
+#include "ComponentTransform.h"
+
+#include "animation/AnimController.h"
+#include "importers/AnimationImporter.h"
+
+Hachiko::ComponentAnimation::ComponentAnimation(GameObject* container) : Component(Type::ANIMATION, container)
+{
+    controller = new AnimController();
```
Comment (baransrc) [inline Source/src/components/ComponentAnimation.cpp:11]: Let's rename this to AnimationController for consistency next time.

File: Source/src/components/ComponentAnimation.h [hunk 2]
```diff
@@ -0,0 +1,39 @@
+#pragma once
+
+#include "Component.h"
+#include <resources/ResourceAnimation.h>
```
Comment (baransrc) [inline Source/src/components/ComponentAnimation.h:4]:
```suggestion
#include "resources/ResourceAnimation.h"
```
We use "" for internal headers and <> for external libraries.

File: Source/src/components/animation/AnimController.h [hunk 3]
```diff
@@ -0,0 +1,25 @@
+#pragma once
+
+#include "Math.h"
```
Comment (baransrc) [inline Source/src/components/animation/AnimController.h:3]: If this is external, let's use angular bracets.
```suggestion
#include <Math.h>
```

File: Source/src/importers/AnimationImporter.h [hunk 4]
```diff
@@ -0,0 +1,22 @@
+#pragma once
+
+#include "assimp/scene.h"
```
Comment (baransrc) [inline Source/src/importers/AnimationImporter.h:3]: For external library headers, we use angular braces.
```suggestion
#include <assimp/scene.h>
```

File: Source/src/resources/ResourceAnimation.h [hunk 5]
```diff
@@ -0,0 +1,82 @@
+#pragma once
+
+#include "resources/Resource.h"
+#include <unordered_map>
+
+#include <memory>
+#include <MathGeoLibFwd.h>
+
+using namespace std;
```
Comment (baransrc) [inline Source/src/resources/ResourceAnimation.h:9]: Let's delete this using statement, as we do `std::` everywhere already.
```suggestion
```

File: Source/src/resources/ResourceAnimation.h [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+            unsigned int num_rotations = 0;
+        };
+
+        struct Node
+        {
+            Node() = default;
+            Node(const std::string name, unsigned int index_parent) : name(name), index_parent(index_parent)
+            {
+                ;
+            }
```
Comment (baransrc) [inline Source/src/resources/ResourceAnimation.h:36]:
```suggestion
            Node(const std::string name, unsigned int index_parent) : name(name), index_parent(index_parent) {}
```

File: Source/src/resources/ResourceTexture.h [hunk 7]
```diff
@@ -58,7 +58,7 @@ namespace Hachiko
         unsigned mag_filter = 0x2601;
         unsigned wrap = 0x2900;
 
-        byte* data{};
+        ::byte* data{};
```
Comment (baransrc) [inline Source/src/resources/ResourceTexture.h:61]: What's the meaning of `::byte*` here?

**Review by baransrc — CHANGES_REQUESTED** [review]
Thanks for the work, I've requested some small changes, then we are good to go.

File: Source/src/resources/ResourceTexture.h (hunk 7 above)
Comment (monikamnc) [reply Source/src/resources/ResourceTexture.h:61]: I had the same question and this is the solution of why is writed in that way
https://developercommunity.visualstudio.com/t/error-c2872-byte-ambiguous-symbol/93889?viewtype=solutions

Comment (baransrc) [reply Source/src/resources/ResourceTexture.h:61]: Noicee

**Review by baransrc — APPROVED** [review]

File: Source/src/components/ComponentAnimation.cpp (hunk 1 above)
Comment (baransrc) [reply Source/src/components/ComponentAnimation.cpp]: Renamed.

**Pull request merged**